src/minecraft/client.rs

```diff
@@ -1,33 +1,43 @@
 use super::{connection::MinecraftConnection, proto, Packet};
 use anyhow::Result;
-use futures::pin_mut;
 use mcproto_rs::{protocol::State, types::CountedArray};
 use std::net::SocketAddr;
 
 use std::sync::Arc;
-use tokio::sync::{
-    mpsc::{error::TryRecvError, Receiver},
+use tokio::{sync::{
+    mpsc::{self, error::TryRecvError, Receiver, Sender},
     Mutex,
-};
+}, runtime::Runtime, task::JoinHandle, join};
+
+use super::server::PacketListener;
 
 /// Error text when the client is not connected to a server.
 const SERVER_NONE_ERROR: &str = "Not connected to server.";
 /// Error when the client recieves a Packet it was not expecting.
 const WRONG_PACKET_ERROR: &str = "Recieved an unexpected packet.";
 
 /// A Minecraft client, a wrapper for the utilities required to connect to a Minecraft server and perform various activities in said server.
-pub struct Client {
+pub struct MinecraftClient {
     /// The socket address of the server the client will connect to.
     address: SocketAddr,
+    /// Tokio runtime
+    runtime: Arc<Mutex<Runtime>>,
```

Comment:
I believe that this arc mutex can be avoided by using a tokio [runtime Handle](https://docs.rs/tokio/latest/tokio/runtime/struct.Handle.html).